### DISCUSSION TITLE
earth_search Sentinel-2: Missing datatake metadata file

### DISCUSSION THREAD

**Discussion by remi-braun** [CS-SI/eodag#404] [Q&A] [answered]
Hello,

When downloading Sentinel-2 products from `earth_search` provider (at least L2A), the datatake metadata file (`MTD_MSIL1C.xml` or `MTD_MSIL2A.xml`) is missing, preventing the user to read some very important information such as:
- the true name of the product
- the datatake sensing time
- the processing baseline number
- the quantification values of the bands
- ...

What would be the best way to retrieve them somehow ?
Or am I missing something ?

**Comment by sbrunato** [comment]
Hello Remi, these files should not be missing.
They are identified using `S2_PROD_MTD_REGEX` in https://github.com/CS-SI/eodag/blob/d9f16f411fdf5eb6ad752f984782dd06ed882dc8/eodag/plugins/download/aws.py#L114-L117
and saved in SAFE product root directory with:
https://github.com/CS-SI/eodag/blob/d9f16f411fdf5eb6ad752f984782dd06ed882dc8/eodag/plugins/download/aws.py#L790-L795

If you have an example of product where this file is missing, can you please give me its id so I can check it on my side ?

**Reply by remi-braun** [reply]
I think this comes from the fact that `earth_search` retrieved Sentinel-2 data from SentinelHub bucket, where those files are already missing (not an eodag issue, this is why I put it in discussion).
You can download almost any S2L2A from `earth_search` (i.e. S2B_MSIL2A_20191005T104029_N0213_R008_T31UEP_20191005T135415)

**Reply by remi-braun** [reply]
I got some errors (not affecting the downloading), maybe it has the same root: 
![2022-02-22_16h15_03](https://user-images.githubusercontent.com/67311115/155162035-303e0ee2-e375-4ae9-b6c9-bfef74328da3.png)

**Comment by sbrunato** [comment, marked as answer]
Missing `manifest.safe` and metadata for some products was due to a malformed regex in metadata extraction configuration for `earth_search`.
See Pull Request [GH-405](https://github.com/CS-SI/eodag/pull/405)

**Reply by remi-braun** [reply]
Ok I'm relieved then 👍
